codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java

```diff
@@ -246,6 +230,24 @@ protected B codec(Http2ConnectionDecoder decoder, Http2ConnectionEncoder encoder
     }
 
     /**
+     * Returns if HTTP headers should be validated according to
+     * <a href="https://tools.ietf.org/html/rfc7540#section-8.1.2.6">RFC 7540, 8.1.2.6</a>.
+     */
+    protected boolean isValidateHeaders() {
+        return validateHeaders != null ? validateHeaders : true;
+    }
+
+    /**
+     * Sets if HTTP headers should be validated according to
+     * <a href="https://tools.ietf.org/html/rfc7540#section-8.1.2.6">RFC 7540, 8.1.2.6</a>.
+     */
+    protected B validateHeaders(boolean validateHeaders) {
+        enforceNonCodecConstraints("validateHeaders");
```

Comment:
similarly, could we change `enforceConstraint` to something like `checkFieldNotSet`?